File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/MetricMonitor.java
package com.sequenceiq.periscope.monitor;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Component;

import com.sequenceiq.periscope.domain.Cluster;
import com.sequenceiq.periscope.monitor.evaluator.MetricEvaluator;

@Component
@ConditionalOnProperty(prefix = "periscope.enabledAutoscaleMonitors.metric-monitor", name = "enabled", havingValue = "true")
public class MetricMonitor extends ClusterMonitor {

    @Override
    public String getIdentifier() {
        return "metric-monitor";
    }

    @Override
    public String getTriggerExpression() {
        return MonitorUpdateRate.METRIC_UPDATE_RATE_CRON;
    }

    @Override
    public Class<?> getEvaluatorType(Cluster cluster) {
        return MetricEvaluator.class;
    }
}

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/MetricMonitor.kt
package com.sequenceiq.periscope.monitor

import java.util.Collections

import org.springframework.stereotype.Component

import com.sequenceiq.periscope.domain.Cluster
import com.sequenceiq.periscope.monitor.evaluator.EvaluatorContext
import com.sequenceiq.periscope.monitor.evaluator.MetricEvaluator

@Component
class MetricMonitor : AbstractMonitor(), Monitor {

    override val identifier: String
        get() = "metric-monitor"

    override val triggerExpression: String
        get() = MonitorUpdateRate.METRIC_UPDATE_RATE_CRON

    override val evaluatorType: Class<Any>
        get() = MetricEvaluator::class.java

    override fun getContext(cluster: Cluster): Map<String, Any> {
        return Collections.singletonMap<String, Any>(EvaluatorContext.CLUSTER_ID.name, cluster.id)
    }
}
